h2o-automl/src/test/java/ai/h2o/automl/targetencoding/TargetEncoderMojoWriterTest.java

```diff
@@ -0,0 +1,64 @@
+package ai.h2o.automl.targetencoding;
+
+import hex.ModelStubs;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import water.Scope;
+import water.TestUtil;
+import water.fvec.Frame;
+
+import java.io.File;
+import java.io.FileOutputStream;
+
+import static org.junit.Assert.*;
+
+public class TargetEncoderMojoWriterTest extends TestUtil implements ModelStubs {
+
+  @BeforeClass
+  public static void stall() { stall_till_cloudsize(1); }
+
+  Frame trainFrame = parse_test_file("./smalldata/gbm_test/titanic.csv");
+
+  @Test
+  public void writeModelToZipFile() throws Exception{
+    Scope.enter();
+    try {
+      TestModel.TestParam p = new TestModel.TestParam();
```

Comment:
why do you need a TestModel class here, why not train a regular TEModel?

Reply:
@honzasterba I agree that in current state of PR I can refactor this and probably get rid of the ModelStubs class. Just as a comment, you never know  the evolution of the code and at the moment of writing this test real TEModel(its params) might not be in a reliable state. Secondly, It is up to an author to decide to what extent his test is integration or unit. ;)

Reply:
I am not sure what are you trying to say, but could you please answer my question. What is the motivation behind TestModel, why cannot you use TEModel directly here as other simmilar test for other algos?

Reply:
ModelStubs is only providing the extended Parameter class - imho not necessary - would be better to just use regular TargetEncoderParams

Reply:
(in this particular test)

Reply:
I actually agreed to refactor this :) But I still believe that I have all rights to have any stubs for whatever reasons even with just one usage. In unit testing it is actually a requirement to use stubs and mocks for anything that you don't want to test in a particular test.... even if there are real components that could be used instead.

Reply:
I am not saying we need to refactor this. I did not understand why is this necessary and I am just asking for an explanation.